packages/process/process_monitor.pony

```diff
@@ -295,20 +305,37 @@ actor ProcessMonitor
   fun _child(
     path: String,
     argp: Array[Pointer[U8] tag],
-    envp: Array[Pointer[U8] tag])
+    envp: Array[Pointer[U8] tag],
+    wdir: (FilePath | None))
   =>
     """
-    We are now in the child process. We redirect STDIN, STDOUT and STDERR
-    to their pipes and execute the command. The command is executed via
-    execve which does not return on success, and the text, data, bss, and
-    stack of the calling process are overwritten by that of the program
-    loaded. We've set the FD_CLOEXEC flag on all file descriptors to ensure
-    that they are all closed automatically once @execve gets called.
+    We are now in the child process. If a working directory was passed to
+    the child, a chdir is attempted. We redirect STDIN, STDOUT and STDERR to
+    their pipes and execute the command. The command is executed via execve which
+    does not return on success, and the text, data, bss, and stack of the
+    calling process are overwritten by that of the program loaded. We've set the
+    FD_CLOEXEC flag on all file descriptors to ensure that they are all closed
+    automatically once @execve gets called.
     """
     ifdef posix then
+      match wdir
+      | let d: FilePath =>
+        let dir: Pointer[U8] tag = d.path.cstring()
+
+        if 0 > @chdir[I32](dir) then
+          let errno: I32 = @pony_os_errno()
+          let strerror = @strerror[Pointer[U8]](errno)
+          let size = @strlen[USize](strerror)
+          let len = @write[ISize](_errpipe_write, strerror, size)
+          @_exit[None](I32(71)) // EX_OSERR from sysexits.h
```

Comment:
Returning `71` is fine here. I think the number makes sense. But we should also get a `ProcessNotify.failed` with the actual reason for the failure.